prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -252,14 +266,21 @@ def get_storage_controller_data(self) -> Tuple[Dict[str, int], Dict[str, List]]:
         for system_id in system_ids:
             storage_controller_count[system_id] = 0
 
+            # finding storage name used in URI by querying Systems dictionary
+            # eg: if storage root URI is /redfish/v1/Systems/S1/Storages
+            # then storage_name is "Storages"
+            system_dict = self.redfish_obj.get(self.systems_root_uri + system_id).dict
+            storage_name = system_dict["Storage"]["@odata.id"].split("/")[-1]
```

Comment:
Can here raise the KeyError if `Storage` is not in the dict? If not please ignore this comment.

Reply:
As part of the redfish schema specification, the dictionary of resources from the systems API call should have a `Storage` resource that points to the URI for fetching the storage data. Unless the system doesn't have any storage at all which I highly unlikely I believe.

But it's a good thing to think about.